index.html

```diff
@@ -1831,322 +1831,291 @@ <h2>Proof</h2>
-and services can use any other data representation syntax, such as JXD (JSON XDI
-Data, a serialization format for the
-<a href="http://docs.oasis-open.org/xdi/xdi-core/v1.0/csd01/xdi-core-v1.0-csd01.xml">XDI graph model</a>),
-XML, YAML, or CBOR, that is capable of expressing the data model.
-      </p>
-
-    <section>
-      <h2>
-Pure JSON
-      </h2>
-
-        <p>
-The data model, as described in Section <a href="#data-model"></a>, can be
-encoded in Javascript Object Notation (JSON) [[RFC8259]] by mapping property
-values to JSON types as follows:
-        </p>
-
-        <ul>
-          <li>
-Numeric values representable as IEEE754 SHOULD be represented as a Number type.
-          </li>
-          <li>
-Boolean values SHOULD be represented as a Boolean type.
-          </li>
-          <li>
-Sequence value SHOULD be represented as an Array type.
-          </li>
-          <li>
-Unordered sets of values SHOULD be represented as an Array type.
-          </li>
-          <li>
-Sets of properties SHOULD be represented as an Object type.
-          </li>
-          <li>
-Empty values SHOULD be represented as a null value.
-          </li>
-          <li>
-Other values MUST be represented as a String type.
-          </li>
-        </ul>
-    </section>
+    <p>
+All concrete representations of a <a>DID document</a> MUST be serialized using a 
+deterministic mapping that is able to be unambiguously parsed into the data model
+defined in this specification. All serialization methods MUST define rules for 
+the bidirectional translation of a <a>DID document</a> both into and out of the
+representation in question. As a consequence, translation between any two representations
+MUST be done by parsing the source format into a <a>DID document</a> model
+(described in Sections <a href="#data-model"></a> and <a href="#did-documents"></a>)
+and then serializing the <a>DID document</a> model into the target representation. An
+implementation MUST NOT convert between representations without first parsing to a 
+<a>DID document</a> model. 
+    </p>
 
-    <section>
-      <h2>
-JSON-LD
-      </h2>
+    <p>
+Although syntactic mappings are provided for JSON, JSON-LD, and CBOR here, applications
+and services MAY use any other data representation syntax that is capable of expressing 
+the data model, such as XML or YAML.
+    </p>
 
-      <p>
-[[!JSON-LD]] is a JSON-based format used to serialize
-<a href="http://www.w3.org/TR/ld-glossary/#linked-data">Linked Data</a>. The
-syntax is designed to easily integrate into deployed systems already using
-JSON, and provides a smooth upgrade path from JSON to JSON-LD. It is primarily
-intended to be a way to use Linked Data in Web-based programming environments,
-to build interoperable Web services, and to store Linked Data in JSON-based
-storage engines.
-      </p>
+    <p>
+Producers MUST indicate which representation of a document has been used, and consumers
```

Comment:
This normative statement isn't clear how producers indicate which representation of a document has been used. My assumption is that there are two ways -- media type and including `@context`. We might want to state that this should only be done via media type?

Reply:
The media type is part of the metadata, and I would argue the most important part right now. That's why this section needs to be written up. See #203 (which is linked below), the "representation type" is the only one specifically named.

Detecting the presence or absence of a context field, or any other internal field, is an anti pattern and needs to be not just avoided but forbidden by the spec's definition of a consumer. Not only is it fragile with the JSON/JSON-LD case, it completely falls apart as soon as you talk about any other representation type, at all.

Reply:
> Detecting the presence or absence of a context field, or any other internal field, is an anti pattern and needs to be not just avoided but forbidden by the spec's definition of a consumer. Not only is it fragile with the JSON/JSON-LD case, it completely falls apart as soon as you talk about any other representation type, at all.

Right, I agree with you completely, which is why I think we might want to reword the above from:

> Producers MUST indicate which representation of a document has been used, and consumers

to this:

> Producers MUST indicate representations of this data model via a media type (see Section E...). Consumers MUST determine representations of this data model via the media type (see Section E...).

Reply:
I will not add language to enable content sniffing nor will I sit by to see it added here.